### ISSUE TITLE
Getting weird data when fetching CSV

### ISSUE THREAD

**Issue by zisra** [axios/axios#5326]
### Describe the bug

Hello, I've been fetching a csv file with one redirect. This worked in a previous version (unsure which one) but when I upgraded it did not work as you can see. I got some weird data I don't know how to parse, but I expect a csv file in a text format. 
<img width="847" alt="image" src="https://user-images.githubusercontent.com/100528712/204715993-4629d70d-1a31-4a6a-a1a6-d2aed61ef1cc.png">
I tried this with cross-fetch but it worked
<img width="847" alt="image" src="https://user-images.githubusercontent.com/100528712/204716052-474d411b-476b-47f8-a99c-9412153f4e18.png">
**My code below:**
Axios:
```js
var axios = require("axios")
const { data } = await axios('https://docs.google.com/spreadsheets/d/1b2z_lTIPEVhabnP73Mir3ttOZlqgEoLV9mLE1T-m1Y4/export?format=csv', {
  responseType: 'text'
});
console.log(data)
```
cross-fetch: 
```js
var fetch = require("cross-fetch")
const data = await fetch('https://docs.google.com/spreadsheets/d/1b2z_lTIPEVhabnP73Mir3ttOZlqgEoLV9mLE1T-m1Y4/export?format=csv');
console.log(await data.text())
```

### Axios Version

1.2.0

### Node.js Version

18.7.0

### OS

MacOS

```bash
cross-fetch: 3.1.5
```

**Comment by akcgjc007** [comment]
I'm also getting a similar error in MacOS in one of my requests. It was working fine a few minutes ago, suddenly this happened:

```
{
  status: 200,
  statusText: 'OK',
  headers: AxiosHeaders {
    'content-type': 'application/json; charset=utf-8',
    etag: '"16vuqgdg0ej1v8"',
    vary: 'Accept-Encoding',
    date: 'Wed, 30 Nov 2022 07:45:15 GMT',
    connection: 'close',
    'transfer-encoding': 'chunked'
  },
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [ 'xhr', 'http' ],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    env: { FormData: [Function], Blob: null },
    validateStatus: [Function: validateStatus],
    headers: AxiosHeaders {
      Accept: 'application/json, text/plain, */*',
      'User-Agent': 'axios/1.2.0',
      'Accept-Encoding': 'gzip, deflate, br'
    },
    method: 'get',
    url: 'http://localhost:3000/api/pools',
    data: undefined
  },
  request: <ref *1> ClientRequest {
    _events: [Object: null prototype] {
      abort: [Function (anonymous)],
      aborted: [Function (anonymous)],
      connect: [Function (anonymous)],
      error: [Function (anonymous)],
      socket: [Function (anonymous)],
      timeout: [Function (anonymous)],
      finish: [Function: requestOnFinish]
    },
    _eventsCount: 7,
    _maxListeners: undefined,
    outputData: [],
    outputSize: 0,
    writable: true,
    destroyed: true,
    _last: true,
    chunkedEncoding: false,
    shouldKeepAlive: false,
    maxRequestsOnConnectionReached: false,
    _defaultKeepAlive: true,
    useChunkedEncodingByDefault: false,
    sendDate: false,
    _removedConnection: false,
    _removedContLen: false,
    _removedTE: false,
    strictContentLength: false,
    _contentLength: 0,
    _hasBody: true,
    _trailer: '',
    finished: true,
    _headerSent: true,
    _closed: true,
    socket: Socket {
      connecting: false,
      _hadError: false,
      _parent: null,
      _host: 'localhost',
      _closeAfterHandlingError: false,
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 6,
      _maxListeners: undefined,
      _writableState: [WritableState],
      allowHalfOpen: false,
      _sockname: null,
      _pendingData: null,
      _pendingEncoding: '',
      server: null,
      _server: null,
      parser: null,
      _httpMessage: [Circular *1],
      [Symbol(async_id_symbol)]: 70787,
      [Symbol(kHandle)]: null,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: null,
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(kCapture)]: false,
      [Symbol(kSetNoDelay)]: false,
      [Symbol(kSetKeepAlive)]: true,
      [Symbol(kSetKeepAliveInitialDelay)]: 60,
      [Symbol(kBytesRead)]: 819,
      [Symbol(kBytesWritten)]: 172,
      [Symbol(RequestTimeout)]: undefined
    },
    _header: 'GET /api/pools HTTP/1.1\r\n' +
      'Accept: application/json, text/plain, */*\r\n' +
      'User-Agent: axios/1.2.0\r\n' +
      'Accept-Encoding: gzip, deflate, br\r\n' +
      'Host: localhost:3000\r\n' +
      'Connection: close\r\n' +
      '\r\n',
    _keepAliveTimeout: 0,
    _onPendingData: [Function: nop],
    agent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      defaultPort: 80,
      protocol: 'http:',
      options: [Object: null prototype],
      requests: [Object: null prototype] {},
      sockets: [Object: null prototype],
      freeSockets: [Object: null prototype] {},
      keepAliveMsecs: 1000,
      keepAlive: false,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      scheduling: 'lifo',
      maxTotalSockets: Infinity,
      totalSocketCount: 1,
      [Symbol(kCapture)]: false
    },
    socketPath: undefined,
    method: 'GET',
    maxHeaderSize: undefined,
    insecureHTTPParser: undefined,
    path: '/api/pools',
    _ended: true,
    res: IncomingMessage {
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 4,
      _maxListeners: undefined,
      socket: [Socket],
      httpVersionMajor: 1,
      httpVersionMinor: 1,
      httpVersion: '1.1',
      complete: true,
      rawHeaders: [Array],
      rawTrailers: [],
      aborted: false,
      upgrade: false,
      url: '',
      method: null,
      statusCode: 200,
      statusMessage: 'OK',
      client: [Socket],
      _consuming: false,
      _dumped: false,
      req: [Circular *1],
      responseUrl: 'http://localhost:3000/api/pools',
      redirects: [],
      [Symbol(kCapture)]: false,
      [Symbol(kHeaders)]: [Object],
      [Symbol(kHeadersCount)]: 14,
      [Symbol(kTrailers)]: null,
      [Symbol(kTrailersCount)]: 0,
      [Symbol(RequestTimeout)]: undefined
    },
    aborted: false,
    timeoutCb: null,
    upgradeOrConnect: false,
    parser: null,
    maxHeadersCount: null,
    reusedSocket: false,
    host: 'localhost',
    protocol: 'http:',
    _redirectable: Writable {
      _writableState: [WritableState],
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      _options: [Object],
      _ended: true,
      _ending: true,
      _redirectCount: 0,
      _redirects: [],
      _requestBodyLength: 0,
      _requestBodyBuffers: [],
      _onNativeResponse: [Function (anonymous)],
      _currentRequest: [Circular *1],
      _currentUrl: 'http://localhost:3000/api/pools',
      [Symbol(kCapture)]: false
    },
    [Symbol(kCapture)]: false,
    [Symbol(kBytesWritten)]: 0,
    [Symbol(kEndCalled)]: true,
    [Symbol(kNeedDrain)]: false,
    [Symbol(corked)]: 0,
    [Symbol(kOutHeaders)]: [Object: null prototype] {
      accept: [Array],
      'user-agent': [Array],
      'accept-encoding': [Array],
      host: [Array]
    },
    [Symbol(kUniqueHeaders)]: null
  },
  data: "\x1F�\b\x00\x00\x00\x00\x00\x00\x13��_K\x1C1\x10\x00���>d2�I&oU\x0B�����S)�?\x13��'�B���\x1B���\x1E+�@.O�ٙ���$����\x10ǫ\x1B�\x1Bj\x1A�\x10\x7F�\x0E'gu�\x03c�\x16\x12\x13J\x0EFم$�\x15��!�z���=��ιb��Z�\b�Zȅ\x18�\x12W����(B\x0Ez���R�!I��\x1Cr��\x00�J�✆�R�\x02XD}��|�:���qq�N�oL/���\x18\x0F���>��$�/o.�!����'��ɥ^\x15�\x7F\x07c��&\x1A᥆�T�\x17,� ��\tS���*\r�`����l0v7\x1Au9�xx�ذ �0z��z��YG8 2=\x06�6Z~�e�T�\n" +
    '�ڔ��^Ulc��ْ�D��\x00eK�|�c������y\x1A�J��?,>\x1D̉y�`F\f�C\f\x05g�E�YL&b}��\x16�%�-I�ȭ\x04�\\���\x05�uS��\x14����C�\x7F;\\���Z\x04\x1C�\x19�U�8��Y��\x15#\n' +
    `\f�z"vYS0�ǡ�'��\n` +
    "疵���lҦO^�Ұ�\x1A�%�껱\x06���d�U'\\�\x0B�4/*診�� 4�nMo:\x07\x1EC'��\x15�j��=���҉H\x10\x17�[d��X��a�%\x07������М����@�\x10����v#6�^��N�����`�e$7�\x152׾S�:\x1Ae&��\x01��+�K@���8�0���\x0Fn����\x06���y�\x17HL\x19�t\t\x00\x00"
}
```

**Comment by nick22985** [comment]
Dupe of #5298

**Comment by zisra** [comment]
Setting Accept-Encoding to * fixed it. Thanks

**Comment by zisra** [comment]
Question: Is this a feature or a bug? It's very annoying to fix and I'm sure new members will have trouble with this as well

**Comment by akcgjc007** [comment]
@zisra defintely a bug bro. I had to use fetch() in my project due to this error.

**Comment by tm1000** [comment]
Fixed in 1.2.1

**Comment by zisra** [comment]
Seems to be fixed now
